lib/drivers/tandemTslimDriver.js

```diff
@@ -1374,6 +1375,28 @@ module.exports = function (config) {
     return records;
   };
 
+  var buildNewDayRecords = function(data, records) {
+    var newDayRecords = filterLogEntries([PUMP_LOG_RECORDS.LID_NEW_DAY], data.log_records);
+
+    for (var b = 0; b < newDayRecords.length; ++b) {
+      var event = newDayRecords[b];
+      var utcTime = sundial.applyTimezone(event.deviceTime, cfg.timezone).toISOString();
+      if(event.commanded_basal_rate !== undefined) {
+        // new day event; breaks up flat-rate basals
+        var postrecord = cfg.builder.makeScheduledBasal()
+            .with_deviceTime(event.deviceTime)
+            .with_rate(event.commanded_basal_rate)
+            .set('index', event.index)
+            .with_conversionOffset(0)
+            .with_timezoneOffset(0)
+            .with_time(utcTime); //new day events should always occur at midnight, so don't shift with time changes
```

Comment:
New day events are midnight local time, _not_ midnight UTC. Use bootstrapping via the tzoUtil as per usual. If that causes some sort of issue with the flat-rate basal data you have, let's troubleshoot together.